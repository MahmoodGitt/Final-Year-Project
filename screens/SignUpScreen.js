import React, { useState } from 'react';

// Firebase Packages
import auth from '../firebase/config';
import { createUserWithEmailAndPassword } from 'firebase/auth';

// Design Packages
import {
	View,
	Text,
	TouchableOpacity,
	TextInput,
	Platform,
	StyleSheet,
	StatusBar,
	ScrollView,
	Alert,
} from 'react-native';
import * as Animatable from 'react-native-animatable';
// import LinearGradient from 'expo-linear-gradient';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import Feather from 'react-native-vector-icons/Feather';
import { Card } from 'react-native-paper';

const SignUpScreen = ({ navigation }) => {
	// Email field holds the user's email details, and a mutator field that stores new email values
	// const [email, setEmail] = useState('');
	// Password field holds the user's email details, and a mutator field that stores new password values
	// const [password, setPassword] = useState('');

	const [data, setData] = useState({
		username: '',
		password: '',
		confirm_password: '',
		secureTextEntry: true,
		user_textChange: false,
		confirm_secureTextEntry: true,
	});

	/**
	 * This function creates a user account and stores account details in Firebase. This is a Firebase API that passes
	 * Firebase configuration file, new user's email address and password.
	 */
	const handleSignUp = () => {
		createUserWithEmailAndPassword(auth, data.username, data.password)
			.then((userCredential) => {
				// Signed in
				const user = userCredential.user;
				// ...
				console.log('username', user.email, 'added to Firebase');
			})
			.catch((error) => {
				const errorCode = error.code;
				const errorMessage = error.message;
				console.log('code:', errorCode);
				console.log('message', errorMessage);
			});
	};

	/**
	 * This function checks the length of the username. Usernames are expected to belong to the user's
	 * organisatrion such as university. Therefore, only organisitonal emails are allowed to be registered
	 * @param {String} text
	 */
	const user_textChange = (text) => {
		// if (text.length !== 7) {
		// setData({
		// ...data,
		// username: text,
		// 		user_textChange: true,
		// });
		// } else {
		// 	setData({
		// 	// 	...data,
		// 	// 	username: text,
		// 	// 	user_textChange: false,
		// 	// });
		// }
	};

	const handleValidUser = (text) => {
		// if (text.includes('@edu')) {
		// 	// setData({
		// 	// 	...data,
		// 	// 	isValidUserName: true,
		// 	// 	user_textChange: true,
		// 	// });
		// 	// console.log('pass');
		// } else {
		// // 	setData({
		// // 		...data,
		// // 		isValidUserName: false,
		// // 		user_textChange: false,
		// // 	});
		// // 	// console.log('fail');
		// // }
	};

	const handlePasswordChange = (val) => {
		// setData({
		// 	...data,
		// 	password: val,
		// });
	};

	const handleConfirmPasswordChange = (val) => {
		// setData({
		// 	...data,
		// 	confirm_password: val,
		// });
	};

	const updateSecureTextEntry = () => {
		// setData({
		// 	...data,
		// 	secureTextEntry: !data.secureTextEntry,
		// });
	};

	const updateConfirmSecureTextEntry = () => {
		// setData({
		// 	...data,
		// 	confirm_secureTextEntry: !data.confirm_secureTextEntry,
		// });
	};

	return (
		<View style={styles.container}>
			<Card>
				<Card.Cover
					style={{ height: 200 }}
					source={{ uri: 'https://picsum.photos/700' }}
				/>
			</Card>
			<View style={styles.header}>
				<Text style={styles.text_header}>Register Now!</Text>
			</View>
			<Animatable.View animation="fadeInUpBig" style={styles.footer}>
				<View>
					<Text style={styles.text_footer}>Username</Text>
					<View style={styles.action}>
						<FontAwesome name="user-o" color="#05375a" size={20} />
						<TextInput
							placeholder="Your Username"
							style={styles.textInput}
							autoCapitalize="none"
							onChangeText={(text) => (data.username = text)}
							onEndEditing={(endText) =>
								handleValidUser(endText.nativeEvent.text)
							}
						/>
						{data.user_textChange ? (
							<Animatable.View animation="bounceIn">
								<Feather name="check-circle" color="green" size={20} />
							</Animatable.View>
						) : null}
					</View>

					<Text
						style={[
							styles.text_footer,
							{
								marginTop: 35,
							},
						]}
					>
						Password
					</Text>
					<View style={styles.action}>
						<Feather name="lock" color="#05375a" size={20} />
						<TextInput
							placeholder="Your Password"
							secureTextEntry={data.secureTextEntry ? true : false}
							style={styles.textInput}
							autoCapitalize="none"
							onChangeText={(text) => (data.password = text)}
						/>
						<TouchableOpacity onPress={updateSecureTextEntry}>
							{data.secureTextEntry ? (
								<Feather name="eye-off" color="grey" size={20} />
							) : (
								<Feather name="eye" color="grey" size={20} />
							)}
						</TouchableOpacity>
					</View>

					<Text
						style={[
							styles.text_footer,
							{
								marginTop: 35,
							},
						]}
					>
						Confirm Password
					</Text>
					<View style={styles.action}>
						<Feather name="lock" color="#05375a" size={20} />
						<TextInput
							placeholder="Confirm Your Password"
							secureTextEntry={data.confirm_secureTextEntry ? true : false}
							style={styles.textInput}
							autoCapitalize="none"
							onChangeText={(val) => handleConfirmPasswordChange(val)}
						/>
						<TouchableOpacity onPress={updateConfirmSecureTextEntry}>
							{data.secureTextEntry ? (
								<Feather name="eye-off" color="grey" size={20} />
							) : (
								<Feather name="eye" color="grey" size={20} />
							)}
						</TouchableOpacity>
					</View>
					<View style={styles.button}>
						<TouchableOpacity style={styles.signIn} onPress={handleSignUp}>
							{/* <LinearGradient
								colors={['#08d4c4', '#01ab9d']}
								style={styles.logIn}
							> */}
							<Text style={styles.textSign}>Sign In</Text>
							{/* </LinearGradient> */}
						</TouchableOpacity>

						<TouchableOpacity
							onPress={() => navigation.goBack()}
							style={[
								styles.signIn,
								{
									borderColor: '#009387',
									borderWidth: 1,
									marginTop: 15,
								},
							]}
						>
							<Text
								style={[
									styles.textSign,
									{
										color: '#009387',
									},
								]}
							>
								Go back
							</Text>
						</TouchableOpacity>
					</View>
				</View>
			</Animatable.View>
		</View>
	);
};

export default SignUpScreen;

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: '#009387',
	},
	header: {
		flex: 0.3,
		justifyContent: 'center',
		alignItems: 'center',
	},
	footer: {
		flex: 3,
		backgroundColor: '#fff',
		borderTopLeftRadius: 30,
		borderTopRightRadius: 30,
		paddingHorizontal: 20,
		paddingVertical: 30,
	},
	text_header: {
		color: '#fff',
		fontWeight: 'bold',
		fontSize: 30,
	},
	text_footer: {
		color: '#05375a',
		fontSize: 18,
	},
	action: {
		flexDirection: 'row',
		marginTop: 10,
		borderBottomWidth: 1,
		borderBottomColor: '#f2f2f2',
		paddingBottom: 5,
	},
	actionError: {
		flexDirection: 'row',
		marginTop: 10,
		borderBottomWidth: 1,
		borderBottomColor: '#FF0000',
		paddingBottom: 5,
	},
	textInput: {
		flex: 1,
		marginTop: Platform.OS === 'ios' ? 0 : -12,
		paddingLeft: 10,
		color: '#05375a',
	},
	errorMsg: {
		color: '#FF0000',
		fontSize: 14,
	},
	button: {
		alignItems: 'center',
		marginTop: 50,
	},
	signIn: {
		width: '100%',
		height: 50,
		justifyContent: 'center',
		alignItems: 'center',
		borderRadius: 10,
	},
	textSign: {
		fontSize: 18,
		fontWeight: 'bold',
	},
});
