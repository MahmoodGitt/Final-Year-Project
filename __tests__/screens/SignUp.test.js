/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';

import SignUpScreen from '../../screens/SignUpScreen';

describe('Sign up Component', () => {
	it('Testing if Sign up Screen Component loads up', () => {
		const tree = renderer.create(<SignUpScreen />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
