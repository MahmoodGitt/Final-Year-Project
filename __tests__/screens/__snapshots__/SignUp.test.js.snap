// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign up Component Testing if Sign up Screen Component loads up 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#009387",
      "flex": 1,
    }
  }
>
  <View
    forwardedRef={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "opacity": 0,
        "paddingHorizontal": 20,
        "paddingVertical": 30,
        "transform": Array [
          Object {
            "translateY": 500,
          },
        ],
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <RCTScrollView>
        <View>
          <View>
            <Text
              style={
                Object {
                  "color": "#05375a",
                  "fontSize": 18,
                  "marginTop": 35,
                }
              }
            >
              Student Name
            </Text>
            <View
              style={
                Object {
                  "borderBottomColor": "#f2f2f2",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "marginTop": 10,
                  "paddingBottom": 5,
                }
              }
            >
              <Text />
              <TextInput
                allowFontScaling={true}
                autoCapitalize="none"
                onChangeText={[Function]}
                onEndEditing={[Function]}
                placeholder="Your Name"
                rejectResponderTermination={true}
                style={
                  Object {
                    "flex": 1,
                    "marginTop": 0,
                    "paddingLeft": 10,
                  }
                }
                underlineColorAndroid="transparent"
              />
            </View>
            <View>
              <Text
                style={
                  Object {
                    "color": "#05375a",
                    "fontSize": 18,
                    "marginTop": 35,
                  }
                }
              >
                University
              </Text>
              <View
                style={
                  Object {
                    "borderBottomColor": "#f2f2f2",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "marginTop": 10,
                    "paddingBottom": 5,
                  }
                }
              >
                <Text />
                <View
                  style={
                    Object {
                      "marginLeft": 5,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignSelf": "stretch",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                      testID="ios_touchable_wrapper"
                    >
                      <View
                        pointerEvents="box-only"
                      >
                        <TextInput
                          allowFontScaling={true}
                          editable={false}
                          rejectResponderTermination={true}
                          style={
                            Array [
                              undefined,
                              Object {
                                "color": "#c7c7cd",
                              },
                            ]
                          }
                          testID="text_input"
                          underlineColorAndroid="transparent"
                          value="Select University"
                        />
                      </View>
                    </View>
                    <Modal
                      hardwareAccelerated={false}
                      onOrientationChange={[Function]}
                      supportedOrientations={
                        Array [
                          "portrait",
                          "landscape",
                        ]
                      }
                      testID="ios_modal"
                      transparent={true}
                      visible={false}
                    >
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "flex": 1,
                            "opacity": 1,
                          }
                        }
                        testID="ios_modal_top"
                      />
                      <View
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#f8f8f8",
                              "borderTopColor": "#dedede",
                              "borderTopWidth": 1,
                              "flexDirection": "row",
                              "height": 45,
                              "justifyContent": "space-between",
                              "paddingHorizontal": 10,
                              "zIndex": 2,
                            },
                            undefined,
                          ]
                        }
                        testID="input_accessory_view"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "flexDirection": "row",
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "opacity": 1,
                              }
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                    "borderColor": "#a1a1a1",
                                    "borderRightWidth": 1.5,
                                    "borderTopWidth": 1.5,
                                    "height": 15,
                                    "width": 15,
                                  },
                                  undefined,
                                  Object {
                                    "marginLeft": 11,
                                    "transform": Array [
                                      Object {
                                        "translateY": 4,
                                      },
                                      Object {
                                        "rotate": "-45deg",
                                      },
                                    ],
                                  },
                                  undefined,
                                  Object {},
                                ]
                              }
                            />
                          </View>
                          <View
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "opacity": 1,
                              }
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                    "borderColor": "#a1a1a1",
                                    "borderRightWidth": 1.5,
                                    "borderTopWidth": 1.5,
                                    "height": 15,
                                    "width": 15,
                                  },
                                  undefined,
                                  Object {
                                    "marginLeft": 22,
                                    "transform": Array [
                                      Object {
                                        "translateY": -5,
                                      },
                                      Object {
                                        "rotate": "135deg",
                                      },
                                    ],
                                  },
                                  undefined,
                                  Object {},
                                ]
                              }
                            />
                          </View>
                        </View>
                        <View
                          accessible={true}
                          focusable={true}
                          hitSlop={
                            Object {
                              "bottom": 4,
                              "left": 4,
                              "right": 4,
                              "top": 4,
                            }
                          }
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                          testID="done_button"
                        >
                          <View
                            testID="needed_for_touchable"
                          >
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#007aff",
                                    "fontSize": 17,
                                    "fontWeight": "600",
                                    "paddingRight": 11,
                                    "paddingTop": 1,
                                  },
                                  undefined,
                                  Object {},
                                ]
                              }
                              testID="done_text"
                            >
                              Done
                            </Text>
                          </View>
                        </View>
                      </View>
                      <View
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#d0d4da",
                              "justifyContent": "center",
                            },
                            Object {
                              "height": 215,
                            },
                            undefined,
                          ]
                        }
                      >
                        <View>
                          <RNCPicker
                            items={
                              Array [
                                Object {
                                  "label": "Select University",
                                  "testID": undefined,
                                  "textColor": undefined,
                                  "value": "Select",
                                },
                                Object {
                                  "label": "Apple",
                                  "testID": undefined,
                                  "textColor": undefined,
                                  "value": "apple",
                                },
                                Object {
                                  "label": "Banana",
                                  "testID": undefined,
                                  "textColor": undefined,
                                  "value": "banana",
                                },
                              ]
                            }
                            onChange={[Function]}
                            selectedIndex={0}
                            style={
                              Array [
                                Object {
                                  "height": 216,
                                },
                                undefined,
                              ]
                            }
                            testID="ios_picker"
                          />
                        </View>
                      </View>
                    </Modal>
                  </View>
                </View>
              </View>
            </View>
            <Text
              style={
                Object {
                  "color": "#05375a",
                  "fontSize": 18,
                  "marginTop": 35,
                }
              }
            >
              Email
            </Text>
            <View
              style={
                Object {
                  "borderBottomColor": "#f2f2f2",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "marginTop": 10,
                  "paddingBottom": 5,
                }
              }
            >
              <Text />
              <TextInput
                allowFontScaling={true}
                autoCapitalize="none"
                onChangeText={[Function]}
                onEndEditing={[Function]}
                placeholder="Your Email"
                rejectResponderTermination={true}
                style={
                  Object {
                    "flex": 1,
                    "marginTop": 0,
                    "paddingLeft": 10,
                  }
                }
                underlineColorAndroid="transparent"
              />
            </View>
            <Text
              style={
                Object {
                  "color": "#05375a",
                  "fontSize": 18,
                  "marginTop": 35,
                }
              }
            >
              Password
            </Text>
            <View
              style={
                Object {
                  "borderBottomColor": "#f2f2f2",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "marginTop": 10,
                  "paddingBottom": 5,
                }
              }
            >
              <Text />
              <TextInput
                allowFontScaling={true}
                autoCapitalize="none"
                onChangeText={[Function]}
                onEndEditing={[Function]}
                placeholder="Your Password"
                rejectResponderTermination={true}
                secureTextEntry={true}
                style={
                  Object {
                    "flex": 1,
                    "marginTop": 0,
                    "paddingLeft": 10,
                  }
                }
                underlineColorAndroid="transparent"
              />
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text />
              </View>
            </View>
            <Text
              style={
                Object {
                  "color": "#05375a",
                  "fontSize": 18,
                  "marginTop": 35,
                }
              }
            >
              Confirm Password
            </Text>
            <View
              style={
                Object {
                  "borderBottomColor": "#f2f2f2",
                  "borderBottomWidth": 1,
                  "flexDirection": "row",
                  "marginTop": 10,
                  "paddingBottom": 5,
                }
              }
            >
              <Text />
              <TextInput
                allowFontScaling={true}
                autoCapitalize="none"
                onChangeText={[Function]}
                onEndEditing={[Function]}
                placeholder="Confirm Your Password"
                rejectResponderTermination={true}
                secureTextEntry={true}
                style={
                  Object {
                    "flex": 1,
                    "marginTop": 0,
                    "paddingLeft": 10,
                  }
                }
                underlineColorAndroid="transparent"
              />
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text />
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "marginTop": 15,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "borderColor": "#009387",
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "height": 50,
                    "justifyContent": "center",
                    "marginTop": 50,
                    "opacity": 1,
                    "width": "50%",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 18,
                        "fontWeight": "bold",
                      },
                      Object {
                        "color": "#009387",
                      },
                    ]
                  }
                >
                  Sign In
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
